pkg/apis/core/v1/defaults.go

```diff
@@ -216,6 +217,11 @@ func SetDefaults_Pod(obj *v1.Pod) {
 			}
 		}
 	}
+
+	if resourcehelper.IsPodLevelResourcesSet(obj, utilfeature.DefaultFeatureGate.Enabled(features.PodLevelResources)) {
```

Comment:
I think this warrants a comment.  To a naive reader it looks like: "If it is already set, then set it", which isn't REALLY what's happening.  Also, I don't think passing the gate down helps readability.  Maybe simpler:

```
if utilfeature.DefaultFeatureGate.Enabled(features.PodLevelResources) {
    defaultPodResources(obj)
}
```

and

```
func defaultPodResources(obj *v1.Pod) {
    // We only populate defaults when the pod-level resources are partly specified already.
    if !resourcehelper.IsPodLevelResourcesSet(obj) {
        return
    }
```